code/core/src/test/java/com/adobe/marketing/mobile/internal/util/MapUtilsTests.java

```diff
@@ -287,6 +287,92 @@ public void testGetFnv1aHash_WithMaskMatchingNoKeys() {
         assertEquals(expectedHash, hash);
     }
 
+    @Test
+    public void testGetFnv1aHash_NullAndEmptyMapValuesPresent_WithAllKeysInMask() {
+        // setup
+        final Map<String, Object> map =
+                new HashMap<String, Object>() {
+                    {
+                        put("a", "1");
+                        put("b", "2");
+                        put("c", "");
+                        put("d", null);
+                    }
+                };
+        // test
+        final long hash = MapUtilsKt.convertMapToFnv1aHash(map, new String[] {"a", "b", "c", "d"});
+        // verify flattened map string "a:1b:2"
+        final long expectedHash = 3371500665L;
+        assertEquals(expectedHash, hash);
+    }
+
+    @Test
+    public void
+            testGetFnv1aHash_NullAndEmptyMapValuesPresent_WithKeysForNullAndEmptyValuesInMask() {
+        // setup
+        final Map<String, Object> map =
+                new HashMap<String, Object>() {
+                    {
+                        put("a", "1");
+                        put("b", "2");
+                        put("c", "");
+                        put("d", null);
+                    }
+                };
+        // test
+        final long hash = MapUtilsKt.convertMapToFnv1aHash(map, new String[] {"c", "d"});
+        // verify 0 / no hash generated due to mask keys being present for null and empty values
+        final long expectedHash = 0;
+        assertEquals(expectedHash, hash);
+    }
+
+    @Test
+    public void testGetFnv1aHash_NullAndEmptyMapValuesPresentInInnerMap_WithAllKeysInMask() {
+        // setup
+        final Map<String, Object> inner =
+                new HashMap<String, Object>() {
+                    {
+                        put("a", "1");
+                        put("b", "2");
+                        put("c", "");
+                        put("d", null);
+                    }
+                };
+        final Map<String, Object> map =
+                new HashMap<String, Object>() {
+                    {
+                        put("inner", inner);
+                    }
+                };
+        // test
+        final long hash =
+                MapUtilsKt.convertMapToFnv1aHash(map, new String[] {"inner.a", "inner.b"});
+        // verify flattened map string "inner.a:1inner.b:2"
+        final long expectedHash = 3328417429L;
+        assertEquals(expectedHash, hash);
+    }
+
+    @Test
+    public void testGetFnv1aHash_NullAndEmptyMapValuesPresent_WithNoMask() {
```

Comment:
looks good.  given the newly introduced method acts recursively, let's make sure we add a test which covers the recursive behavior as well